datumaro/plugins/synthetic_data/utils.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from pathlib import Path
+import os.path as osp
+
+import cv2 as cv
+import numpy as np
+import requests
+
+
+class IFSFunction:
+    def __init__(self, rng, prev_x, prev_y):
+        self.function = []
+        self.xs, self.ys = [prev_x], [prev_y]
+        self.select_function = []
+        self.cum_proba = 0.0
+        self._rng = rng
+
+    def set_param(self, params, proba, weights=None):
+        if weights is not None:
+            params = list(np.array(params) * np.array(weights))
+
+        self.function.append(params)
+        self.cum_proba += proba
+        self.select_function.append(self.cum_proba)
+
+    def calculate(self, iterations):
+        rand = [self._rng.random() for _ in range(iterations)]
```

Comment:
It seems unnecessary to pregenerate these ahead of time. You could just put `rand = self._rng.random()` at the beginning of the `for` loop.

Reply:
done